packages/common/src/selling-partner-api-error.ts

```diff
@@ -0,0 +1,83 @@
+import {AxiosError} from 'axios'
+import {URL} from 'url'
+
+interface AmazonError {
+  messsage: string;
+  code: string;
+}
+
+interface Request {
+  pathname?: string;
+  domain?: string;
+  method?: string;
+  body?: string | Record<any, any>;
+  queries?: Record<any, any>;
+}
+
+interface Response {
+  errors: AmazonError[];
+  status: number;
+  headers: Record<string, string>;
+}
+
+export class SellingPartnerApiError extends Error {
+  public resource?: string
+  public action?: string
+  public response?: Response
+  public request?: Request
+  public apiName?: string
+  public apiVersion?: string
+
+  constructor(error: AxiosError) {
+    super()
+    let body = error.config.data
+    const url = error.config.url ? new URL(error.config.url) : null
+
+    try {
+      body = JSON.parse(body)
+    } catch {}
+
+    this.resource = error.config.url
+    this.action = error.config.method
+    this.request = {
+      body,
+      method: error.config.method
+    }
+
+    if (url) {
+      this.request.pathname = url.pathname
+      this.request.queries = url.searchParams && Object.fromEntries(url.searchParams.entries())
+      this.request.domain = url.host
+
+      const [apiName, apiVersion] = url.pathname.split('/').slice(1)
+      this.apiName = apiName
+      this.apiVersion = apiVersion
+    }
+
+    if (error.response) {
+      this.response = {
+        errors: error.response.data?.errors,
+        status: error.response.status,
+        headers: error.response.headers
+      }
```

Comment:
Why not expose the full response?

Maybe `SellingPartnerApiError` could inherit `AxiosError` and we could just add the properties we need on top?